### PR TITLE
Fix documentation in `haddock-api`

### PR DESCRIPTION
**Pull request by harpocrates**
* fix some misplaced docs in `haddock-api`
  * make CI check that docs build

### CODE REVIEW

File: haddock-api/src/Haddock/Backends/LaTeX.hs [hunk 1]
```diff
@@ -444,11 +444,11 @@ ppLPatSig doc docnames ty unicode
...
                -> DocForDecl DocName  -- ^ documentation
-               -> ( LaTeX             -- ^ first-line (no-argument docs only)
-                  , LaTeX             -- ^ first-line (argument docs only)
-                  , LaTeX             -- ^ type prefix (argument docs only)
+               -> ( LaTeX             --   first-line (no-argument docs only)
+                  , LaTeX             --   first-line (argument docs only)
+                  , LaTeX             --   type prefix (argument docs only)
                   )
-               -> Bool                -- ^ unicode
+               -> Bool                --   unicode
```
Comment (sjakobi) [inline haddock-api/src/Haddock/Backends/LaTeX.hs]: That bit of haddocks shouldn't be problematic no?

Comment (harpocrates) [reply haddock-api/src/Haddock/Backends/LaTeX.hs]: Thanks for noticing! Phew.

**Comment by harpocrates** [comment]
> Also: Do we have any plans to make Haddock accept docs on subfields of function parameters?

YES! But I'm gonna wait until after Hi Haddock. We're going to have to store docs in a better way that just as a `Map Name Doc` and `Map Name (IntMap Doc)`...

**Comment by sjakobi** [comment]
> YES! But I'm gonna wait until after Hi Haddock. We're going to have to store docs in a better way that just as a `Map Name Doc` and `Map Name (IntMap Doc)`...

Sounds great! :)

**Pull request merged**